code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubClient.java

```diff
@@ -0,0 +1,165 @@
+package io.cattle.platform.iaas.api.auth.github;
+
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+import io.cattle.platform.json.JsonMapper;
+import io.cattle.platform.util.type.CollectionUtils;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.fluent.Request;
+
+import com.netflix.config.DynamicStringProperty;
+
+public class GithubClient {
+
+    private JsonMapper jsonMapper;
+
+    private static final String GITHUB_REQUEST_TOKEN = "code";
+    private static final String CLIENT_ID = "client_id";
+    private static final String CLIENT_SECRET = "client_secret";
+
+    public static final DynamicStringProperty GITHUB_URL = ArchaiusUtil.getString("api.auth.github.url");
+
+    private static final DynamicStringProperty GITHUB_CLIENT_ID = ArchaiusUtil.getString("api.auth.github.client.id");
+    private static final DynamicStringProperty GITHUB_CLIENT_SECRET = ArchaiusUtil.getString("api.auth.github.client.secret");
+    private static final DynamicStringProperty GITHUB_USER_INFO_URL = ArchaiusUtil.getString("api.auth.github.user.url");
+    private static final DynamicStringProperty GITHUB_ORG_INFO_URL = ArchaiusUtil.getString("api.auth.github.org.url");
+    private static final DynamicStringProperty GITHUB_UNAUTH_USER_URL = ArchaiusUtil.getString("api.github.user.url");
+    private static final DynamicStringProperty GITHUB_UNAUTH_ORG_URL = ArchaiusUtil.getString("api.github.org.url");
+
+    public Map<String, Object> getAccessToken(String code) {
+        List<RequestParam> requestData = new ArrayList<>();
+
+        if (StringUtils.isEmpty(code)) {
+            throw new ClientVisibleException(ResponseCodes.UNPROCESSABLE_ENTITY, "code=Missing Required");
+        }
+
+        requestData.add(new RequestParam(CLIENT_ID, GITHUB_CLIENT_ID.get()));
```

Comment:
You don't need this RequestParam class.  `org.apache.http.message.BasicNameValuePair` already exists for this purpose.